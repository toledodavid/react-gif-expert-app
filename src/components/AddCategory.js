import React, { useState } from 'react';

const AddCategory = () => {

  const [inputValue, setInputValue] = useState('Type something');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
  }

  return (
    <form onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder={inputValue}
        onChange={handleInputChange}
      />
    </form>
  );
}

export default AddCategory;