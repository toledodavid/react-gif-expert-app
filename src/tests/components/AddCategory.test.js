import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('Tests for <AddCategory /> component', () => {

  const setCategories = () => {};
  const wrapper = shallow(<AddCategory setCategories={setCategories} />);

  test('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('It should change input value', () => {
    const input = wrapper.find('input');
    const value = 'Mario Bros';
    input.simulate('change', {target: {value}});

    expect(wrapper.find('p').text().trim()).toBe(value);
  });

});