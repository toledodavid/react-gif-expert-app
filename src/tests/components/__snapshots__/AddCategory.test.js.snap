// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <AddCategory /> component It should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder=""
    type="text"
  />
</form>
`;
